bigchaindb/common/election.py

```diff
@@ -0,0 +1,248 @@
+# Copyright BigchainDB GmbH and BigchainDB contributors
+# SPDX-License-Identifier: (Apache-2.0 AND CC-BY-4.0)
+# Code is Apache-2.0 and docs are CC-BY-4.0
+
+import base58
+
+from bigchaindb import backend
+from bigchaindb.common.exceptions import (InvalidSignature,
+                                          MultipleInputsError,
+                                          InvalidProposer,
+                                          UnequalValidatorSet,
+                                          DuplicateTransaction)
+from bigchaindb.tendermint_utils import key_from_base64
+from bigchaindb.common.crypto import (public_key_from_ed25519_key)
+from bigchaindb.common.transaction import Transaction
+from bigchaindb.common.schema import (_validate_schema,
+                                      TX_SCHEMA_COMMON,
+                                      TX_SCHEMA_CREATE)
+
+
+class Election(Transaction):
+
+    # NOTE: this transaction class extends create so the operation inheritance is achieved
+    # by setting an ELECTION_TYPE and renaming CREATE = ELECTION_TYPE and ALLOWED_OPERATIONS = (ELECTION_TYPE,)
+    ELECTION_TYPE = None
+    # the name of the mongodb table managed by the election
+    DB_TABLE = None
+    # the model for votes issued by the election
+    VOTE_TYPE = None
+    # Election Statuses:
+    ONGOING = 'ongoing'
+    CONCLUDED = 'concluded'
+    INCONCLUSIVE = 'inconclusive'
+    # Vote ratio to approve an election
+    ELECTION_THRESHOLD = 2 / 3
+
+    @classmethod
+    def get_validator_change(cls, bigchain, height=None):
+        """Return the latest change to the validator set
+
+        :return: {
+            'height': <block_height>,
+            'asset': {
+                'height': <block_height>,
+                'validators': <validator_set>,
+                'election_id': <election_id_that_approved_the_change>
+            }
+        }
+        """
```

Comment:
I think this doc string is wrong? 

```

        :return: {
                'height': <block_height>,
                'validators': <validator_set>,
                'election_id': <election_id_that_approved_the_change>
        }
```

Reply:
Nope. I checked it. It's correct.

Reply:
Don't think so, `bigchain.get_validator_change(height)` calls [`backend.query.get_validator_set(self.connection, height)`](https://github.com/bigchaindb/bigchaindb/blob/cb22557771c2b75fa24607f58a1526e802fbe96f/bigchaindb/lib.py#L430) which returns the validator object of the following format,

```python
{'height': <block_height>,
 'validators': <validator_set>,
 'election_id': <election_id_that_approved_the_change>
}
```